src/foundry/client/data/documents/cards.d.ts

```diff
@@ -18,44 +16,46 @@ declare global {
     /**
      * Provide a thumbnail image path used to represent this document.
      */
-    get thumbnail(): string | null;
+    get thumbnail(): string;
 
     /**
-     * The Card documents within this stack which are able to be drawn.
+     * The Card documents within this stack which are available to be drawn.
      */
-    get availableCards(): InstanceType<ConfiguredDocumentClassForName<"Card">>[];
+    get availableCards(): Card[];
 
     /**
      * The Card documents which belong to this stack but have already been drawn.
      */
-    get drawnCards(): InstanceType<ConfiguredDocumentClassForName<"Card">>[];
+    get drawnCards(): Card[];
 
     /**
      * Returns the localized Label for the type of Card Stack this is
      */
     get typeLabel(): string;
 
+    /**
+     * Can this Cards document be cloned in a duplicate workflow?
+     */
+    get canClone(): boolean;
+
     static override createDocuments(
-      data: Array<
-        | ConstructorDataType<foundry.data.CardsData>
-        | (ConstructorDataType<foundry.data.CardsData> & Record<string, unknown>)
-      >,
+      data: Array<ConstructorDataType<CardsData> | (ConstructorDataType<CardsData> & Record<string, unknown>)>,
```

Comment:
This can probably be simplified even more, by using `BaseCards.ConstructorData`. We would then either need to introduce a generic parameter on the `Cards` class for the different system types or leave it open and add a union of all known types as a default to the `ConstructorData`. I think once we are done, we won't need `ConstructorDataType` at all anymore.

The same goes for the other occurrences. We can check each time whether it's `ConstructorData`, `UpdateData` or `Properties`.

Reply:
With that said, should I take further action for this PR?

Reply:
Yes please.